remote-source/main.go

```diff
@@ -0,0 +1,277 @@
+	Path     string `yaml:"path,omitempty"`
+	Branch   string `yaml:"branch,omitempty"`
+}
+
+type Environment struct {
+	Region          string  `yaml:"region"`
+	Organization    string  `yaml:"organization"`
+	StackName       string  `yaml:"stackName"`
+	BaseProject     Project `yaml:"baseProject"`
+	PlatformProject Project `yaml:"platformProject"`
+	AppProject      Project `yaml:"appProject"`
+	DataProject     Project `yaml:"dataProject"`
+}
+
+func main() {
+	// Determine mode of operation; default is to refresh/update
+	destroy := false
+	argsWithoutProg := os.Args[1:]
+	if len(argsWithoutProg) > 0 {
+		if argsWithoutProg[0] == "destroy" {
+			destroy = true
+		}
+	}
+
+	// Get configuration data
+	data, err := os.ReadFile("config.yaml")
+	if err != nil {
+		fmt.Printf("Error reading configuration file: %v\n", err)
+		os.Exit(1)
+	}
+	var env Environment
+	err = yaml.Unmarshal(data, &env)
+	if err != nil {
+		fmt.Printf("Error parsing configuration information: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Set up context
+	ctx := context.Background()
+
+	// Call the setupStack function to set up each of the four stacks
+	// Set up base stack
+	baseStack, err := setupStack(ctx, env.Organization, env.StackName, env.BaseProject)
+	if err != nil {
+		fmt.Printf("Error encountered setting up %s stack: %v\n", env.BaseProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Set up platform stack
+	platformStack, err := setupStack(ctx, env.Organization, env.StackName, env.PlatformProject)
+	if err != nil {
+		fmt.Printf("Error encountered setting up %s stack: %v\n", env.PlatformProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Set up data stack
+	dataStack, err := setupStack(ctx, env.Organization, env.StackName, env.DataProject)
+	if err != nil {
+		fmt.Printf("Error encountered setting up %s stack: %v\n", env.DataProject.Name, err)
+		os.Exit(1)
+	}
+
+	// Set up application stack
+	appStack, err := setupStack(ctx, env.Organization, env.StackName, env.AppProject)
+	if err != nil {
+		fmt.Printf("Error encountered setting up %s stack: %v\n", env.AppProject.Name, err)
+		os.Exit(1)
+	}
+
+	// If destroy is true, call deleteStack function to destroy stacks
+	// The order of operations is important; app, platform, data, base
+	if destroy {
+		_, err := deleteStack(ctx, appStack, env.AppProject.Name)
+		if err != nil {
+			fmt.Printf("Error deleting %s stack: %v\n", env.AppProject.Name, err)
+			os.Exit(1)
+		}
+		_, err = deleteStack(ctx, platformStack, env.PlatformProject.Name)
+		if err != nil {
+			fmt.Printf("Error deleting %s stack: %v\n", env.PlatformProject.Name, err)
+			os.Exit(1)
+		}
+		_, err = deleteStack(ctx, dataStack, env.DataProject.Name)
+		if err != nil {
+			fmt.Printf("Error deleting %s stack: %v\n", env.DataProject.Name, err)
+			os.Exit(1)
+		}
+		_, err = deleteStack(ctx, baseStack, env.BaseProject.Name)
+		if err != nil {
+			fmt.Printf("Error deleting %s stack: %v\n", env.BaseProject.Name, err)
+			os.Exit(1)
+		}
+		os.Exit(0)
```

Comment:
I think this reads better, and it does the same thing.
```suggestion
		return
```